monitoring/rid_qualifier/aircraft_state_replayer.py

```diff
@@ -0,0 +1,146 @@
+            uss = usses[state_data_index]  
+
+
+            test_id = str(uuid.uuid4())
+            test_injection_path = '/tests/{test_id}'.format(test_id=test_id)
+            
+            disk_reference_time_raw = current_disk_rid_state_data['reference_time']
+            disk_reference_time = arrow.get(disk_reference_time_raw)
+
+            current_disk_rid_state_data['reference_time'] = test_reference_time.isoformat()
+            updated_timestamps_telemetry = []            
+            
+            timestamp_offset = test_start_offset - disk_reference_time
+                        
+            for telemetry_id, flight_telemetry in enumerate(current_disk_rid_state_data['flight_telemetry']['states']):
+
+                timestamp = (arrow.get(flight_telemetry['timestamp']) + timestamp_offset)
+
+                
+                flight_telemetry['timestamp'] = timestamp.isoformat()
+                updated_timestamps_telemetry.append(flight_telemetry)
+                        
+            flight_details =  current_disk_rid_state_data['flight_details']
+            operator_details = current_disk_rid_state_data['operator_details']
+            
+            operator_location = OperatorLocation(lat = operator_details['location']['lat'], lng = operator_details['location']['lng'])            
+            rid_flight_details = RIDFlightDetails(operator_id = operator_details['operator_id'], operator_location = operator_location, operation_description = flight_details['operation_description'] , serial_number = flight_details['serial_number'], registration_number = operator_details['registration_number'])
+            test_flight_details = TestFlightDetails(effective_after = test_start_offset_isoformat,details = [rid_flight_details])
+            test_flight = TestFlight(injection_id = str(uuid.uuid4()), telemetry = updated_timestamps_telemetry, details_responses = test_flight_details)
+            test_flight_deserialized = self.make_json_compatible(test_flight)
+            requested_flights.append(test_flight_deserialized)
+            test_payload = TestPayload(test_id = test_id, requested_flights = requested_flights)
+            test_payload_data_metadata = DeliverablePayloads(injection_path = test_injection_path, injection_payloads = test_payload)
+            all_test_payloads.append(test_payload_data_metadata)
+        
+        return all_test_payloads
+
+
+class TestHarness():
+    ''' A class to submit Aircraft RID State to the USS test endpoint '''
+
+    def __init__(self, auth_spec:str, injection_base_url:str):
+        
+        auth_adapter = make_auth_adapter(auth_spec)
+        self.uss_session = DSSTestSession(injection_base_url, auth_adapter)
+
+    def submit_test(self,uss_session:DSSTestSession, payload:DeliverablePayloads) -> None:
+        
+        response = uss_session.put(url = payload.injection_path, json=payload.injection_payloads, scope = ' '.join([rid.SCOPE_RID_QUALIFIER_INJECT ]))
```

Comment:
The JSON in `payload.injection_payloads` does not match the schema required by this endpoint

Reply:
This is now fixed, I have renamed the DeliverablePayloads -> DeliverablePayload and made changes to the variable naming.

Reply:
Let me know if you prefer the plural, `List[DeliverablePayload]` reads better (to me)

Reply:
I'm most concerned with consistency, so if it's consistent with singular throughout, that sounds great.  I agree that it does seem odd to have multiple "payloads" if the content will all be delivered in a single call.